Build/QIR_validation.ps1

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = 'Stop'
+
+& "$PSScriptRoot/set-env.ps1"
+$all_ok = $True
+
+function Build-One {
+    param(
+        $project
+    );
+
+    $project_directory = $(Split-Path $project -Parent)
+    $project_name = $(Split-Path $project -LeafBase)
+
+    pushd $project_directory
+
+    Write-Host "##[info]Building $project..."
+    dotnet build $project `
+        -c $Env:BUILD_CONFIGURATION `
+        -v $Env:BUILD_VERBOSITY `
+        --no-dependencies `
+        /property:DefineConstants=$Env:ASSEMBLY_CONSTANTS `
+        /property:Version=$Env:ASSEMBLY_VERSION `
+        /property:QirGeneration="true"
+
+    if  ($LastExitCode -ne 0) {
+        Write-Host "##vso[task.logissue type=error;]Failed to build $project."
+        $script:all_ok = $False
+    } else {
+        qir-cli `
```

Comment:
Is it required that this tool is already installed? If so, checking for the command earlier and providing an actionable error message when it is missing can help both internal and external developers working with this script. To check if a command is missing:

```powershell
if (-not (Get-Command qir-cli -ErrorAction SilentlyContinue)) {
    Write-Host "##[error]The qir-cli command is missing; you can install it by running `TODO`.";
}
```

Reply:
Yah this is a really good idea. I've added a check like this.